src/type-system.ts

```diff
@@ -354,6 +361,33 @@ export const ElysiaType = {
 			})
 			.Encode((value) => value) as any as TNumber
 	},
+	Integer: (property?: NumberOptions): TInteger => {
+		const schema = Type.Integer(property)
+
+		return t
+			.Transform(
+				t.Union(
+					[
+						t.String({
+							format: 'integer',
+							default: 0
```

Comment:
i know `Numeric` does this too but shouldn't this be `property.default`?

Reply:
this seems to break other tests which I feel are out of scope for this PR:

```ts
✓ Replace Schema Type > replace object properties in Union
 96 |                           {
 97 |                                   from: t.Number(),
 98 |                                   to: () => t.Numeric()
 99 |                           }
100 |                   )
101 |           ).toMatchObject(
          ^
error: expect(received).toMatchObject(expected)

  {
    [Symbol(TypeBox.Kind)]: "Object",
    properties: {
      id: {
        [Symbol(TypeBox.Kind)]: "Union",
        [Symbol(TypeBox.Transform)]: {
          Decode: [Function],
          Encode: [Function],
        },
        anyOf: [
          {
            [Symbol(TypeBox.Kind)]: "String",
+           default: 0,
-           default: 1,
            format: "numeric",
+           title: "hello",
            type: "string",
          },
          {
            [Symbol(TypeBox.Kind)]: "Number",
            default: 1,
            title: "hello",
            type: "number",
          }
        ],
        default: 1,
        title: "hello",
      },
      name: {
        [Symbol(TypeBox.Kind)]: "String",
        type: "string",
      },
    },
    required: [
      "id",
      "name"
    ],
    type: "object",
  }

- Expected  - 1
+ Received  + 2

```